scipy/sparse/_csr.py

```diff
@@ -122,6 +122,70 @@ def tobsr(self, blocksize=None, copy=True):
 
     tobsr.__doc__ = _spbase.tobsr.__doc__
 
+
+    def _broadcast_to(self, shape, copy=False):        
+        if self.shape == shape:
+            return self.copy() if copy else self
+        
+        shape = check_shape(shape, allow_nd=(self._allow_nd))
+
+        both_are_1d = len(self.shape) == 1 and len(shape) == 1
+        if both_are_1d:
+            if self.shape != (1,):
+                raise ValueError(f'current shape {self.shape} cannot be'
+                             f' broadcast to new shape {shape}')
+
+            self.sum_duplicates()
+            if self.nnz == 0: # array has no non zero elements
+                return self.__class__(shape, dtype=self.dtype, copy=False)
+            
+            N = shape[0]
+            data = np.full(N, self.data[0])
+            indices = np.arange(0,N)
+            indptr = np.array([0, N])
+            return self.__class__((data, indices, indptr), shape=shape, copy=False)
+
+        if len(self.shape) == 1:
+            self = self.__class__(self.reshape(self._shape_as_2d))
+
+        old_shape = self.shape
```

Comment:
No need to reshape self. We should just use `self._shape_as_2d`.
```suggestion
        # treat 1D as a 2D row
        old_shape = self._shape_as_2d
```